frontend_ta/src/components/CDFrame.js

```diff
@@ -3,22 +3,196 @@ import "../styles/iframe.css"
+  }
+  function WaitStart() {
+    now = Date.now();
+    if (now < data.startingDate) {
+
+      timeforphase = data.startingDate - now;
+      setDays(Math.floor(timeforphase / (1000 * 60 * 60 * 24)));
+      setHours(Math.floor((timeforphase / (1000 * 60 * 60)) % 24));
+      setMinutes(Math.floor((timeforphase / 1000 / 60) % 60));
+      setSeconds(Math.floor((timeforphase / 1000) % 60));
+    }
+    else {
+      clearInterval(intervalstopper);
+      InitializeTimer();
+    }
+  }
+  useEffect(() => {
+
+    let page = document.getElementsByTagName("html")[0];
+    page.style.overflowX = "hidden";
+    page.style.overflowY = "hidden";
+    if (location.state == null) {
+      if (sessionStorage.getItem("username") != null) {
+        getData("/api/match/current_match").then((response) => {
+          if (response.status === 200) {
+            response.result.startingDate = response.result.startingDate * 1 // convert startDate from string to integer
+            response.result.phaseOneDuration = response.result.phaseOneDuration * 1000;
+            response.result.phaseTwoDuration = response.result.phaseTwoDuration * 1000;
+            response.result.phaseThreeDuration = response.result.phaseThreeDuration * 1000;
+            data = response.result;
+            setDatarender(data);
+            InitializeTimer(response.result);
+          }
+          else {
+            navigate("/error", { state: { message: response.message } })
+          }
+        }
+
+        );
+      }
+
     }
+    else {
+      location.state.info.startingDate = location.state.info.startingDate * 1; // convert startDate from string to integer
+      location.state.info.phaseOneDuration = location.state.info.phaseOneDuration * 1000;
+      location.state.info.phaseTwoDuration = location.state.info.phaseTwoDuration * 1000;
+      location.state.info.phaseThreeDuration = location.state.info.phaseThreeDuration * 1000;
+      data = location.state.info;
+      setDatarender(data);
+      InitializeTimer(location.state.info);
+    }
+
+
+    return function cleanup() {
+      let page = document.getElementsByTagName("html")[0];
+      page.style.overflowX = "visible";
+      page.style.overflowY = "visible";
+    }
+  }, [])
+
+
+
+
+
+  function DisplayContent() {
+    let timertext;
+    if (cphase == 0) timertext = "game in break phase " + days.toString() + " H: " + hours.toString() + " M: " + minutes.toString() + " S: " + seconds.toString();
+    else if (cphase == 1) timertext = "Phase 1  D: " + days.toString() + " H: " + hours.toString() + " M: " + minutes.toString() + " S: " + seconds.toString();
+    else if (cphase == 2) timertext = "Phase 2  D: " + days.toString() + " H: " + hours.toString() + " M: " + minutes.toString() + " S: " + seconds.toString();
+    else if (cphase == 3) timertext = "Phase 3  D: " + days.toString() + " H: " + hours.toString() + " M: " + minutes.toString() + " S: " + seconds.toString();
+    else if (cphase == 4) timertext = "Ended"
```

Comment:
This part can be improve.
test first the non general cases and then the generale cases to avoid repetitions
``` 
if cphase === 4 => timertext = "Ended"
else
  if cphase === 0 => timertext = "game in break phase 
  else => timertext = "Phase " + cphase
  timertext += "D: " + days.toString() + " H: " + hours.toString() + " M: " + minutes.toString() + " S: " + seconds.toString();
```

And be careful, == should be ===